oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticDocument.java

```diff
@@ -148,4 +166,19 @@ public String build() {
     public String toString() {
         return build();
     }
+
+    public static Collection<Field> newSimilarityFields(String name, Blob value) throws IOException {
+        Collection<Field> fields = new ArrayList<>(1);
+        byte[] bytes = new BlobByteSource(value).read();
+        fields.add(newSimilarityField(name, bytes));
+        return fields;
+    }
+
+    private static Field newSimilarityField(String name, byte[] bytes) {
```

Comment:
For my taste, one of the methods newSimilarityField isn't needed... instead I would inline the logic. That simplifies the code a tiny bit.